### PR TITLE
Few Minor per-utxo assert-semantics re-adds and tweak

### PR DESCRIPTION
**Pull request by TheBlueMatt**
One small tweak which @sdaftuar commented and got missed, and two restore-previous-assert-semantics that I figured would be nice to re-add.

### CODE REVIEW

**Review by gmaxwell — COMMENTED** [review]
utACK (will test)

**Comment by sipa** [comment]
utACK  743bdb5d52f7f7a0bcddfd55bccdac287140b491

File: src/validation.cpp [hunk 1]
```diff
@@ -1118,7 +1118,8 @@ void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, CTxUndo &txund
         txundo.vprevout.reserve(tx.vin.size());
         BOOST_FOREACH(const CTxIn &txin, tx.vin) {
             txundo.vprevout.emplace_back();
-            inputs.SpendCoin(txin.prevout, &txundo.vprevout.back());
+            bool fSpent = inputs.SpendCoin(txin.prevout, &txundo.vprevout.back());
```
Comment (ryanofsky) [inline src/validation.cpp]: In commit "Return a bool in SpendCoin to restore pre-per-utxo assert semantics"

Maybe drop hungarian.

File: src/consensus/tx_verify.cpp [hunk 2]
```diff
@@ -126,7 +126,9 @@ unsigned int GetP2SHSigOpCount(const CTransaction& tx, const CCoinsViewCache& in
     unsigned int nSigOps = 0;
```
Comment (ryanofsky) [inline src/consensus/tx_verify.cpp]: In commit "Restore some assert semantics in sigop cost calculations"

Maybe update commit message to suggest why policy/atmp asserts aren't added back. Not possible there, or just not worth the effort?

**Review by ryanofsky — COMMENTED** [review]
utACK 743bdb5d52f7f7a0bcddfd55bccdac287140b491

It took me a little while to decipher first commit, but just to put it in my own words, the map `mapNextTx.count(txin.prevout)` check that this PR removes would always returns 0 because you can't have two transactions in the mempool spending the same prevout, and the one transaction that had been spending it just got removed out a few lines previously in `RemoveStaged`.

File: src/consensus/tx_verify.cpp (hunk 2 above)
Comment (TheBlueMatt) [inline src/consensus/tx_verify.cpp]: I figured it wasn't worth it - if we hit those errors in consensus code, we'd better crash cause we might fall out of consensus...if we accidentally add something to our mempool that should be out of policy...oh well, not a big deal.

**Review by ryanofsky — COMMENTED** [review]
utACK 9417d7a336e71ee31ce673a6de5abb3d56941204. Has style guide fixes and the updated commit message.

**Comment by sipa** [comment]
utACK 9417d7a336e71ee31ce673a6de5abb3d56941204

**Pull request merged**